post_office/tests/test_mail.py

```diff
@@ -398,6 +399,37 @@ def test_send_bulk_with_faulty_template(self):
         email = Email.objects.get(id=email.id)
         self.assertEqual(email.status, STATUS.failed)
 
+    def test_retry_failed(self):
+        previous_settings = settings.POST_OFFICE
+        setattr(settings, 'POST_OFFICE', {'MAX_RETRIES': 2})
+
+        self.assertEqual(get_max_retries(), 2)
+        self.assertEqual(get_time_delta_to_retry(), timedelta(minutes=15))
+
+
+        email = Email.objects.create(to='to@example.com', from_email='from@example.com', status=STATUS.queued, backend_alias='error')
+        _send_bulk([email], uses_multiprocessing=False)
+        email.refresh_from_db()
+
+        self.assertEqual(int(email.scheduled_time.timestamp()), int(now().__add__(timedelta(minutes=15)).timestamp()))
```

Comment:
By relying on `now()`, you add unpredictable behaviour, and it becomes impossible to step through the testing code using a debugger.